### PR TITLE
Stable trove random password

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
Cherry-pick commits 7d9653f2c68720e30ce3a48d7e9c36fb5f99bbd8 and cdaa9ec2243693d36d1a67bb92271ff974b45e6b into stable/3.0

### CODE REVIEW

File: chef/data_bags/crowbar/template-trove.json [hunk 1]
```diff
@@ -22,6 +23,7 @@
     "trove": {
       "crowbar-revision": 1,
       "crowbar-applied": false,
+      "schema-revision": 101,
```
Comment (vuntz) [inline chef/data_bags/crowbar/template-trove.json]: I think in that branch, we should just bump the version, I assume it's supposed to be 2. Version `>= 100` is for master, usually.

**Comment by nicolasbock** [comment]
+1

**Comment by toabctl** [comment]
+1 (but gating failed)

**Comment by rhafer** [comment]
The gating failure was caused by the recent brokeness in some of the init scripts. I'll trigger a re-run.

**Comment by rhafer** [comment]
+1 once the gate is green

**Comment by rhafer** [comment]
The gating failed because of https://bugs.launchpad.net/manila/+bug/1531667 on. I'll backport that to our liberty packages.

**Comment by rhafer** [comment]
The fix for https://bugs.launchpad.net/manila/+bug/1531667 is now part of our packages. Re-re-running the gate.

**Pull request merged**